commit 8916d6568445bac8dfffa14d83ffbcbae1ce54d2
Author: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
Date:   Fri Jun 24 12:30:32 2016 +0200

    ath9k: add a helper to get the string representation of ath_bus_type
    
    This can be used when the ath_bus_type has to be presented in a log
    message or firmware filename.
    
    Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>

--- a/drivers/net/wireless/ath/ath.h
+++ b/drivers/net/wireless/ath/ath.h
@@ -321,4 +321,10 @@ void _ath_dbg(struct ath_common *common,
 /** Returns string describing opmode, or NULL if unknown mode. */
 const char *ath_opmode_to_string(enum nl80211_iftype opmode);
 
+extern const char *ath_bus_type_strings[];
+static inline const char *ath_bus_type_to_string(enum ath_bus_type bustype)
+{
+	return ath_bus_type_strings[bustype];
+}
+
 #endif /* ATH_H */
--- a/drivers/net/wireless/ath/main.c
+++ b/drivers/net/wireless/ath/main.c
@@ -90,3 +90,10 @@ void ath_printk(const char *level, const
 	va_end(args);
 }
 EXPORT_SYMBOL(ath_printk);
+
+const char *ath_bus_type_strings[] = {
+	[ATH_PCI] = "pci",
+	[ATH_AHB] = "ahb",
+	[ATH_USB] = "usb",
+};
+EXPORT_SYMBOL(ath_bus_type_strings);
